backend/games/word_guesser/ClassicWordle.py

```diff
@@ -0,0 +1,137 @@
+from random import randint
+from typing import Dict, List
+
+
+class ClassicWordle:
+    # maximum number of guesses
+    MAX_GUESSES = 6
+    WORD_LIST = [
+        "adieu",
+        "odium",
+        "shade",
+        "resin",
+        "alert",
+        "haunt",
+        "orate",
+        "media",
+        "blind",
+        "route",
+        "audio",
+        "pause",
+        "alien",
+        "canoe",
+        "plane",
+        "rouse",
+        "fraud",
+        "atone",
+        "raise",
+        "minor",
+    ]
+
+    def __init__(self):
+        # represents the state of the game, index 0 will hold the "goal word"
+        game_board = List[str]
+        guesses = {"guess1": "", "results": []}
+        goal_word = ""
+        guess_count = 0
+
+    # clears any previous activity and randomizes a word to initialize the list
+    def start_game(self) -> List[str]:
+        # clear past game data
+        guesses = {"guess1": "", "results1": []}
+        game_board = [""]
+        goal_word = str
+        guess_count = 0
+
+        # randomize next word
+        new_word_index = randint(0, 19)  # nosec
+        goal_word = WORD_LIST[new_word_index]
+        return game_board
+
+    # clears all previous progress and calls checkWin() to determine final results
+    # will return a victory or failure string
+    def end_game(self) -> str:
+        guesses = {"guess1": "", "results1": []}
+        game_board = List[str]
+        goal_word = ""
+        guess_count = 0
+
+        if check_win():
+            return "You guessed the word!! Nice Job"
+        else:
+            return "You failed to guess the word in 6 guesses. better luck next time."
+
+    """
+        -returns a dict containing the guess and its correctness
+        -list will determine correctness of each letter in guess in the form of :
+            "correct", "in_word", "not_in_word"
+        -updates guesses dict
+    """
+
+    def track_progress(self) -> Dict:
+        progress = {}
+        tmp_keys = ""
+        tmp_results = []
+        for k, v in guesses.items():
```

Comment:
The variable `guesses` is not in scope. It looks like you want the instance variable, so you should use `self.guesses`. That means you also need to update your constructor to set the instance variables:

```python
    def __init__(self):
        ...
        self.guesses = {"guess1": "", "results": []}
        ...
```